### PR TITLE
Game logic

### LINKED ISSUE #4: 2.3 Add game logic

### PR DESCRIPTION
**Pull request by rStrzelczyk98**

### CODE REVIEW

File: src/app/button-group/button-group.component.html [hunk 1]
```diff
@@ -1,4 +1,5 @@
 <div class="d-flex mx-auto mb-4">
   <button class="flex-grow-1 btn bg-success-subtle border border-success text-success text-uppercase">play</button>
-  <button class="flex-grow-1 btn bg-warning-subtle border border-warning text-warning text-uppercase">reset</button>
+  <button class="flex-grow-1 btn bg-warning-subtle border border-warning text-warning text-uppercase" (click)="reset()"
+    [disabled]="gameStatus$ | async">reset</button>
```
Comment (RdN) [inline src/app/button-group/button-group.component.html:4]: You can't reset ongoing game? Why?

File: src/app/service/game-control.service.ts [hunk 2]
```diff
@@ -77,7 +135,67 @@ export class GameControlService {
       .subscribe();
   }
 
+  reload() {
+    this.gameActive.next(true);
+    this.updateRound();
+    this.startWithCross = !this.startWithCross;
+    this.activePlayer.next(this.togglePlayer());
+    this.gameBoard.next(new Array(9).fill(Math.random(), 0, 9));
```
Comment (RdN) [inline src/app/service/game-control.service.ts:143]: Why Math.rand?

File: src/app/service/game-control.service.ts [hunk 3]
```diff
@@ -77,7 +135,67 @@ export class GameControlService {
...
+    this.games$$.next(new ReplaySubject<move>());
+  }
+
+  private findWinner(moves: move[]) {
+    if (this.checkPattern(moves, 'circle')) {
```
Comment (RdN) [inline src/app/service/game-control.service.ts:148]: I would parametrize that method and move circle/cross to params.

**Review by RdN — APPROVED** [review]

File: src/app/service/game-control.service.ts (hunk 2 above)
Comment (rStrzelczyk98) [reply src/app/service/game-control.service.ts:143]: assigning a new Array(9) don't reload the view, so i fill it with random numbers

File: src/app/service/game-control.service.ts (hunk 3 above)
Comment (rStrzelczyk98) [reply src/app/service/game-control.service.ts:148]: done

File: src/app/button-group/button-group.component.html (hunk 1 above)
Comment (rStrzelczyk98) [reply src/app/button-group/button-group.component.html:4]: removed

**Pull request merged**